pyi.py

```diff
@@ -1537,7 +1538,14 @@ def visit_Subscript(self, node: ast.Subscript) -> None:
             self._visit_slice_tuple(node.slice, subscripted_object_name)
         else:
             self.visit(node.slice)
-            if subscripted_object_name in {"tuple", "Tuple"}:
+            if (
+                subscripted_object_name in {"tuple", "Tuple"}
+                and not isinstance(node.slice, ast.Starred)
```

Comment:
It's impossible for `node.slice` to be an instance of `ast.Starred` here:

```pycon
>>> import ast
>>> print(ast.dump(ast.parse('tuple[*Ts]'), indent=2))
Module(
  body=[
    Expr(
      value=Subscript(
        value=Name(id='tuple', ctx=Load()),
        slice=Tuple(
          elts=[
            Starred(
              value=Name(id='Ts', ctx=Load()),
              ctx=Load())],
          ctx=Load()),
        ctx=Load()))],
  type_ignores=[])
```

The py311 tests should continue to work fine if you remove this `isinstance()` check:

```suggestion
```